x-pack/dockerlogbeat/readme.md

```diff
@@ -48,3 +48,11 @@ The location of the logs AND the container base directory in the docker docs is
 You can use this to find the list of plugins running on the host: `runc --root /containers/services/docker/rootfs/run/docker/plugins/runtime-root/plugins.moby/ list`
 
 The logs are in `/var/log/docker`. If you want to make the logs useful, you need to find the ID of the plugin. Back on the darwin host, run `docker plugin inspect $name_of_plugin | grep Id` use the hash ID to grep through the logs: `grep 22bb02c1506677cd48cc1cfccc0847c1b602f48f735e51e4933001804f86e2e docker.*`
+
+
+## Local logs
+
+This plugin fully supports `docker logs`, and it maintains a local log spool in the event that upstream ES connections are down. Unfortunately, due to the limitations in the docker plugin API, we can't "clean up" log files when a container is destroyed. The plugin mounts the `/var/log` directory on the host to write logs. This mount point can be changed via [Docker](https://docs.docker.com/engine/reference/commandline/plugin_set/#change-the-source-of-a-mount). The plugin can also be configured to do a "hard" cleanup and destroy logs when a container stops. To enable this, set the `DESTROY_LOGS_ON_STOP` environment var inside the plugin:
```

Comment:
> This plugin fully supports `docker logs`, and it maintains a local log spool in the event that upstream ES connections are down

This makes it sound like we buffer and continue with buffered logs when ES is up again. But in fact we just maintain a copy of all logs, but Elasticsearch will not be fed from the local copy. With the current implementation we might observe data loss if ES is not reachable (max 3 retries + backoff).